lib/browser/desktop-capturer.js

```diff
@@ -28,45 +28,47 @@ ipcMain.on('ELECTRON_BROWSER_DESKTOP_CAPTURER_GET_SOURCES', function (event, cap
 
   // If the WebContents is destroyed before receiving result, just remove the
   // reference from requestsQueue to make the module not send the result to it.
-  event.sender.once('destroyed', function () {
+  event.sender.once('destroyed', () => {
     request.webContents = null
+    return
   })
 })
 
-desktopCapturer.emit = function (event, name, sources) {
+desktopCapturer.emit = (event, name, sources) => {
   // Receiving sources result from main process, now send them back to renderer.
-  var handledRequest, i, len, ref, ref1, request, result, source, unhandledRequestsQueue
-  handledRequest = requestsQueue.shift(0)
-  result = (function () {
-    var i, len, results
-    results = []
-    for (i = 0, len = sources.length; i < len; i++) {
-      source = sources[i]
+  const handledRequest = requestsQueue.shift(0)
```

Comment:
Do you really need to pass an argument to the `.shift()` call?

Reply:
it's a relic that i moved and not added, if there's a better way to do it i'll phase it out!

Reply:
I mean `Array.prototype.shift()` doesn't accept any arguments, it always returns the first element (and removes it from the array) )

Reply:
I _do_ understand that you are not the author of the code here,
but if you've decided to touch it, let's make it as pretty as possible )

Reply:
rad, changing rn!